pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -399,11 +437,51 @@ func buildInboundTCPFilter(mesh *meshconfig.MeshConfig, attrs attributes, node *
 	}
 }
 
-func addServiceConfig(filterConfigs map[string]*types.Struct, config *mccpb.ServiceConfig) map[string]*types.Struct {
+func addServiceConfig(dir direction, push *model.PushContext, hostname model.Hostname, in *plugin.InputParams,
+	filterConfigs map[string]*types.Struct) map[string]*types.Struct {
 	if filterConfigs == nil {
 		filterConfigs = make(map[string]*types.Struct)
 	}
-	filterConfigs[mixer] = util.MessageToStruct(config)
+	disableCheckCalls := disablePolicyChecks(dir, in.Env.Mesh, in.Node)
+	var key string
+	switch dir {
+	case outbound:
+		if disableCheckCalls {
+			key = strings.Join([]string{string(hostname), "outbound", policyCheckDisable}, "|")
+		} else {
+			key = strings.Join([]string{string(hostname), "outbound", policyCheckEnable}, "|")
+		}
+	case inbound:
+		key = strings.Join([]string{string(hostname), "inbound"}, "|")
```

Comment:
This is awkward. Can you use a struct with three string fields instead of manufacturing a string?

Reply:
Move to plugin.CacheKey: ```
// CacheKey is the key for RDS perRouter Filter Config cache
type CacheKey struct {
	Hostname  string
	Direction string
	Policy    string
}
```